### DISCUSSION TITLE
How do I update the URL in the tracking script?

### DISCUSSION THREAD

**Discussion by lizgw** [plausible/analytics#586] [Self-Hosted Support] [answered]
I have the Plausible dashboard running at `stats.example.com` and added the tracking script tag to my website, but the views aren't registering because the tracking script is trying to send data to localhost instead of the domain I'm running Plausible at. I get this error in the browser console when I visit my website:

`Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://localhost:8000/api/event. (Reason: CORS request did not succeed).`

The script is hosted at `https://stats.example.com/js/plausible.js` but the part at the end is `(window,"http://localhost:8000");` instead of my server's address. Is there a way to change this to something like `stats.example.com`? I have `BASE_URL` set to the proper value in `plausible-conf.env` and I've tried restarting Plausible with `docker-compose down` and `docker-compose up -d` but it doesn't fix the issue. I also removed all docker volumes and images and did a fresh install, but I'm still getting the same issue. Going through the custom domain setup on the web dashboard tells me to point a DNS record to `custom.localhost.`, which doesn't seem right. Setting up a custom domain of `stats.example.com` gives me a script src of `https://stats.example.com/js/index.js` which returns a 404.

Interestingly, in the Plausible web dashboard, the script tag to include in my website has an src value of `http://localhost:8000/js/plausible.js` instead of `https://stats.example.com/js/plausible.js`. I suspect this is because the first time I started docker, I left the default localhost:8000 in the .env file to test it out, but now I can't get it to work at a different URL. I have everything running through an apache reverse proxy but I don't think that's causing the problem. Do you have any ideas on how I can fix this? I'm fairly new to Docker, so I might just be missing something simple. Thanks!

**Comment by ukutaht** [comment, marked as answer]
Thanks for the detailed report.

Most likely this is hapening because the `BASE_URL` needs to have the protocol defined for it to work properly. So instead of `stats.example.com` it should be `https://stats.example.com`.

If this is true I'm sorry to have wasted your time. It's a silly detail which I will make sure to fix for the next release.

**Reply by lizgw** [reply]
That fixed it, thank you! I'm just glad it wasn't one of the more complicated pieces of my server setup, haha. :+1:

**Reply by TheCatster** [reply]
> Thanks for the detailed report.
> 
> Most likely this is hapening because the `BASE_URL` needs to have the protocol defined for it to work properly. So instead of `stats.example.com` it should be `https://stats.example.com`.
> 
> If this is true I'm sorry to have wasted your time. It's a silly detail which I will make sure to fix for the next release.

I'm having the same issue as lizgw, but I do have my `BASE_URL` set with protocol, and I'm not sure what else could be going wrong. My "snippet" shows the proper domain, but the script still shows localhost:8000. Any ideas what that might be?

**Reply by ukutaht** [reply]
The snippet and the script should always be the same... the script is set to cache for 1 hour so make sure your browser cache is disabled and if using a CDN you can try purging the cache.

**Reply by TheCatster** [reply]
I just checked back on this in the morning, and had some other people check, and now I just get a 502 error when the script tries to load (going to the address no longer returns a URL). Not quite sure what I'm doing wrong here, I just put the script in the head of my site. I checked my docker-compose, I can see the dashboard fine, just can't access this script. I am using Cloudflare though, maybe that causes issues?

**Reply by ukutaht** [reply]
:/ that sucks. Weird that you can access the dashboard while getting Bad Gateway from Cloudflare. Can you hit the server without going through CF to see if the script loads?

**Reply by TheCatster** [reply]
That does work! I have it in a Proxmox/pfSense setup (not sure if that is causing an issue?) and when I tunnel 8000 over I get the proper script, showing the right script location. Accessing it through cloudflare however still gives the 502 error, even when bypassing DNS. 🤔
